### ISSUE TITLE
Language url parameter ignored, when requesting a page that does not exist

### ISSUE THREAD

**Issue by bytehead** [contao/contao#456]
**Affected version(s)**
4.7

**Description**  
If you request a page that not exists in a multilanguage environment, e.g. `https://mydomain.tld/en/foobar`, the language parameter `en` is ignored and contao will deliver the 404 content based on your browser language.

`$_GET['language']` is not set here: https://github.com/contao/contao/blob/ed2fec5eaaa84875a9a14b5b90e803a65bccec7c/core-bundle/src/Resources/contao/classes/Frontend.php#L312

In prior versions, this request would be routed via `FrontendIndex.php` and this would call `getPageIdFromUrl()`:
https://github.com/contao/contao/blob/ed2fec5eaaa84875a9a14b5b90e803a65bccec7c/core-bundle/src/Resources/contao/controllers/FrontendIndex.php#L46

which itself sets the `language` property in `$_GET`:
https://github.com/contao/contao/blob/ed2fec5eaaa84875a9a14b5b90e803a65bccec7c/core-bundle/src/Resources/contao/classes/Frontend.php#L259-L282

In 4.7 the Symfony CMF router throws a `NotFoundHttpException` which will be handled by the `PrettyErrorScreenListener` and the listener will call `getRootPageFromUrl()` directly (without executing `getPageIdFromUrl()` upfront).

This causes the language to be missed in `$_GET`.

**How to reproduce**  
Set up a multilanguage installation with configured 404 pages and content.
Request a not existing page in a language that differs from your default browser language. You get the 404 content in your browser's default language instead of the target language.

**Comment by leofeyer** [comment]
@aschempp /cc

**Comment by bytehead** [comment]
Quick (and dirty) fix for existing projects:
```yaml
# services.yaml
services:
    App\EventListener\NotFoundHttpExceptionListener:
        tags:
            - { name: kernel.event_listener, event: kernel.exception, method: onKernelException, priority: 0 }
```

```php
<?php

declare(strict_types=1);

namespace App\EventListener;

use Contao\Frontend;
use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;

class NotFoundHttpExceptionListener
{
    public function onKernelException(GetResponseForExceptionEvent $event): void
    {
        if (!$event->getException() instanceof NotFoundHttpException) {
            return;
        }

        // Call getPageIdFromUrl() to get the language parameter set (see https://github.com/contao/contao/issues/456)
        Frontend::getPageIdFromUrl();
    }
}
```

**Comment by aschempp** [comment]
How can we know a language if there is no valid page? I mean we don't know if the first two chars are supposed to be a language?

**Comment by leofeyer** [comment]
We could check if there is a root page which uses the two chars as language.

**Comment by leofeyer** [comment]
Fixed in a5800915ddedfe7fc16aebb2457e045c99b3b681.
